### PR TITLE
Dispatch v1 flow create to flow hs

### PR DESCRIPTION
**Pull request by timofei-durakov**
- changed worker for flow v1 create request
- fixed compile time warnings

### CODE REVIEW

File: src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/converter/FlowMapper.java [hunk 1]
```diff
@@ -159,6 +160,42 @@ default FlowResponsePayload toFlowResponseOutput(FlowReadResponse r) {
     @Mapping(target = "type", ignore = true)
     FlowRequest toFlowRequest(FlowRequestV2 request);
 
+
+    @Mapping(target = "flowId", source = "id")
+    @Mapping(target = "sourceSwitch", expression = "java(request.getSource().getSwitchDpId())")
```
Comment (surabujin) [inline src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/converter/FlowMapper.java:165]: You don't need eval expression here - instead, you can use `source = "source.switchDpId"`.

File: src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java [hunk 2]
```diff
@@ -170,17 +169,17 @@
...
             logger.error("Can not parse arguments: {}", e.getMessage(), e);
             throw new MessageException(correlationId, System.currentTimeMillis(), ErrorType.DATA_INVALID,
                     e.getMessage(), "Can not parse arguments of the create flow request");
         }
 
         CommandMessage request = new CommandMessage(
-                payload, System.currentTimeMillis(), correlationId, Destination.WFM);
+                flowRequest, System.currentTimeMillis(), correlationId, Destination.WFM);
 
         return messagingChannel.sendAndGet(topic, request)
```
Comment (surabujin) [inline src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java:184]: Should it be placed into `flowHsTopic`?

File: src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java [hunk 3]
```diff
@@ -170,17 +169,17 @@
         final String correlationId = RequestCorrelationId.getId();
         logger.info("Create flow: {}", input);
 
-        FlowCreateRequest payload;
+        FlowRequest flowRequest;
         try {
-            payload = new FlowCreateRequest(new FlowDto(input), input.getDiverseFlowId());
+            flowRequest =  flowMapper.toFlowCreateRequest(input);
```
Comment (surabujin) [inline src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java:174]: Are we ready for a complete switch to H&S create? I.e. remove toggle for a fast rollback on v1 create? (Current mapping have such toggle).

File: src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/converter/FlowMapper.java [hunk 4]
```diff
@@ -159,6 +160,42 @@ default FlowResponsePayload toFlowResponseOutput(FlowReadResponse r) {
...
+    /**
+     * Map FlowCreatePayload.
+     *
+     * @param source {@link FlowCreatePayload} instance.
+     * @return {@link FlowRequest} instance.
+     */
+    default FlowRequest toFlowCreateRequest(FlowCreatePayload source) {
+        FlowRequest target = toFlowRequest(source).toBuilder().type(Type.CREATE).build();
+        DetectConnectedDevicesDto detectConnectedDevicesDto = new DetectConnectedDevicesDto();
+        if (source.getSource().getDetectConnectedDevices() != null) {
```
Comment (surabujin) [inline src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/converter/FlowMapper.java:186]: I would like to have a copy constructor for `DetectConnectedDevicesDto` or mapper method to copy such objects (at least it remove code duplication).

**Comment by rtretyak** [comment]
This will result in a slightly different format of some of the error messages (e.g. bad request). Don't see it as a problem at this point, but maybe sbody can think of any issues related to this

**Comment by rtretyak** [comment]
Due to this change simultaneous v1 create(which is now v2) + v1 delete may cause some rule inconsistencies. Not a blocker as well, just observations

File: src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/converter/FlowMapper.java (hunk 1 above)
Comment (rtretyak) [reply src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/converter/FlowMapper.java]: fixed

File: src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java (hunk 2 above)
Comment (rtretyak) [reply src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java]: fixed

File: src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/converter/FlowMapper.java (hunk 4 above)
Comment (timofei-durakov) [reply src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/converter/FlowMapper.java:186]: in v1 detect info was part of flow endpoint, in v2 it's a separate entity. 
v1: FlowCreatePayload::FlowEndpointPayload::DetectConnectedDevicesPayload(separation based on endpoint level
v2: FlowRequest::DetectConnectedDevicesDto - single object, separation based on field level, 
e.g. srcLldp/dstLldp

Does it make sense to implement such mapper? Could see little gain here.

File: src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java (hunk 3 above)
Comment (timofei-durakov) [reply src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java:174]: It should be just tested, whole point to change worker on permanent basis

**Pull request merged**